File: Harvard/CS50 Intro to Computer Science/pset6/credit/credit.py
from cs50 import get_string

credit = get_string("number: ")
sum1 = 0
sum2 = 0
length = len(credit)

if (length % 2 == 0):    
    
    for i in range(0, length, 2):
        if (len(str(int(credit[i]) * 2)) > 1):
            sum1 += int(str(int(credit[i]) * 2)[0])
            sum1 += int(str(int(credit[i]) * 2)[1])
        elif (len(str(int(credit[i]) * 2)) == 1):
            sum1 += int(credit[i]) * 2
                
    for i in range(1, length, 2):
        sum2 += int(credit[i])

elif (length % 2 != 0):
    
    for i in range(1, length, 2):
        if (len(str(int(credit[i]) * 2)) > 1):
            sum1 += int(str(int(credit[i]) * 2)[0])
            sum1 += int(str(int(credit[i]) * 2)[1])
        elif (len(str(int(credit[i]) * 2)) == 1):
            sum1 += int(credit[i]) * 2
                
    for i in range(0, length, 2):
        sum2 += int(credit[i])
    
total = sum1 + sum2

if (total % 10 != 0):
    print("INVALID")
    
elif (total % 10 == 0):
    if (length == 15 and credit[0] == '3' and (credit[1] == '4' or credit[1] == '7')):
        print("AMEX")
    elif (length == 13 or (length == 16 and credit[0] == "4")):
        print("VISA")
    else:
        print("MASTERCARD")

File: Harvard/CS50 Intro to Computer Science/pset1/credit/credit.c
#include <stdio.h>
#include <cs50.h>

int main(void) 
{
 
    long number = get_long("number: ");
    long numClone = number;
    int everySecondDigit;
    int Sum1 = 0;
    int Sum2 = 0;
    int numDigits = 0;
    
    for (int i = 0; numClone > 0; numClone /= 10)
    {
        numDigits++;
    }
    
    numClone = number;
    
    for (int i = 0; i < numDigits / 2; i++)
    {
        everySecondDigit = numClone % 10;
        numClone /= 100;
        Sum1 += everySecondDigit;
    }
    
    numClone = number;
    numClone /= 10;
    
    for (int i = 0; i < numDigits / 2; i++)
    {
        everySecondDigit = (numClone % 10 * 2);
        numClone /= 100;
        
        if (everySecondDigit / 10 < 0) 
        {
            Sum2 += everySecondDigit;
        }
        else 
        {
            Sum2 += everySecondDigit % 10;
            everySecondDigit /= 10;
            Sum2 += everySecondDigit % 10;
        }
    }
    
    if (numDigits == 15) 
    {
        Sum2 += 3;
    }
    
    if (numDigits == 13)
    {
        Sum2 += 4;
    }
    
    int checkSum = (Sum1 + Sum2) % 10;
    
    if (checkSum != 0)
    {
        printf("INVALID\n");
    }
    else if (numDigits == 15 && ((number / 10000000000000) == 34 || (number / 10000000000000) == 37))
    {
        printf("AMEX\n");
    }
    else if (numDigits == 13)
    {
        printf("VISA\n");
    }
    else if (numDigits == 16 && (number / 100000000000000) > 50 && (number / 100000000000000) < 56) 
    {
        printf("MASTERCARD\n");
    }
    else if (numDigits == 16 && (number / 1000000000000000) == 4)
    {
        printf("VISA\n");
    }
    else 
    {
        printf("INVALID\n");
    }
}